coalib/results/Result.py

```diff
@@ -63,6 +64,11 @@ def __init__(self,
         if severity not in RESULT_SEVERITY.reverse:
             raise ValueError("severity is not a valid RESULT_SEVERITY")
 
+        if(diffs and not isinstance(diffs, dict)):
+            self.diffs = {d.filename: d for d in diffs}
+        else:
+            self.diffs = diffs
```

Comment:
the changes to this file actually belong to an own commit IMO whereas you would have the test changes (https://github.com/coala/coala/pull/2655/commits/8d1e072a5c6aa4cbf7ac21ecd2c77d9eed5c5c13) squashed into this (https://github.com/coala/coala/pull/2655/commits/7797f8816a1a72fd97ea1575eb6cc7a3048b37a5) again. The hunks in this file add new functionality, the others are just changing the use of the Diff.